packages/realm/src/TypeHelpers.ts

```diff
@@ -109,6 +125,20 @@ export function mixedToBinding(realm: binding.Realm, value: unknown): binding.Mi
     return value as binding.MixedArg;
   }
 }
+function isGeoCircle(value: object): value is GeoCircle {
+  return "distance" in value && "center" in value && typeof value["distance"] == "number";
+}
+
+function isGeoBox(value: object): value is GeoBox {
+  return "bottomLeft" in value && "topRight" in value;
+}
+
+function isGeoPolygon(value: object): value is GeoPolygon {
+  return (
+    ("type" in value && value["type"] === "Polygon" && "coordinates" in value && Array.isArray(value["coordinates"])) ||
+    ("outerRing" in value && Array.isArray(value["outerRing"]))
+  );
+}
```

Comment:
Regarding `isGeoCircle()`, `isGeoBox()`, and `isGeoPolygon()`, for the `GeoPoint` properties, would it make sense to create a `isGeoPoint()` function to that each of the aforementioned functions can call to validate those values? Currently, it only checks that the field is defined (e.g. `"center" in value`).

Reply:
I had this function before (and I can retrieve it), but I was thinking that maybe it could be too much to be so strict? ("too much" from the performance point of view)

Reply:
I guess it depends on what behavior Core will have or the errors it will generate if you pass along incorrect types. See what happens with some faulty types 🙂

Reply:
This is a problem for us, not for them, because core requests that we create certain classes. So what will happen is that at some point during the conversion there will be a js exception popping up